src/llscan.cc

```diff
@@ -1242,12 +1243,20 @@ void LLScan::ScanMemoryRanges(FindJSObjectsVisitor& v) {
       for (size_t j = 0; j + addr_size <= loaded;) {
         uint64_t value;
 
-        if (addr_size == 4)
+        if (addr_size == 4) {
           value = *reinterpret_cast<uint32_t*>(&block[j]);
-        else if (addr_size == 8)
+          if (process_byte_order != HOST_BYTE_ORDER) {
+            value = __builtin_bswap32(value);
+          }
+        }
+        else if (addr_size == 8) {
           value = *reinterpret_cast<uint64_t*>(&block[j]);
-        else
+          if (process_byte_order != HOST_BYTE_ORDER) {
+            value = __builtin_bswap64(value);
+          }
+        } else {
```

Comment:
I think it's probably worth raising a separate issue for this. I'm a bit concerned doing a memcpy for every word would eliminate some of the performance gains from reading blocks of memory instead of individual words.